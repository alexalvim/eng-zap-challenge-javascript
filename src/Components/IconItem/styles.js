import styled from 'styled-components';

import Colors from '../../colors';
import Typo from '../../typo'
import Spaces from '../../spaces';

export const ContentWrapper = styled.div`
  align-items: center;
  display: flex;
`;

export const Label = styled.span`
  color: ${Colors.darkerColor};
  font-size: ${Typo.medium};
`;

export const iconStyles = {
  marginRight: Spaces.base,
}