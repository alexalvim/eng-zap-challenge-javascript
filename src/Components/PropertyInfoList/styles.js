import styled from 'styled-components';

import Spaces from '../../spaces';

export const List = styled.ul`
  > li {
    margin-bottom: ${Spaces.base};
  }
`;