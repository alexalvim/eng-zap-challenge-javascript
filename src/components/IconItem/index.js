import React from 'react';

import Colors from '../../colors';
import {
  ContentWrapper,
  Label,
  iconStyles
} from './styles';

export default ({ Icon, label, activePortal }) => (
  <ContentWrapper>
    <Icon
      size={18}
      color={Colors.portalMainColor[activePortal]}
      style={iconStyles} />
    <Label>{label}</Label>
  </ContentWrapper>
)