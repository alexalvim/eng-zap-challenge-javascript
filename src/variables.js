export const PROPERTIES_PER_PAGE = 20;

export const PORTALS = {
  VIVA_REAL: 'VivaReal',
  ZAP: 'Zap',
}

export const BOUDING_BOX_NUMBERS = {
  MIN_LAT: -23.568704,
  MIN_LON: -46.693419,
  MAX_LAT: -23.546686,
  MAX_LON: -46.641146
}