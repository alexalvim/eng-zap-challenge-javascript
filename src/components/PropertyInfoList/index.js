import React from 'react'
import {
  FaBed,
  FaShower,
  FaCar,
  FaRuler,
  FaBuilding
} from "react-icons/fa";

import IconItem from '../IconItem';
import { List } from './styles'

export default ({ property, activePortal }) => (
  <List>
    <li>
      <IconItem
        activePortal={activePortal}
        Icon={FaBed}
        label={`Quartos: ${property.bedrooms}`}/>
    </li>
    <li>
      <IconItem
        activePortal={activePortal}
        Icon={FaShower}
        label={`Banheiros: ${property.bathrooms}`}/>
    </li>
    <li>
      <IconItem
        activePortal={activePortal}
        Icon={FaCar}
        label={`Vagas: ${property.parkingSpaces}`}/>
    </li>
    <li>
      <IconItem
        activePortal={activePortal}
        Icon={FaRuler}
        label={`Área útil: ${property.usableAreas} m²`}/>
    </li>
    {property.address.neighborhood &&
      <li>
        <IconItem
          activePortal={activePortal}
          Icon={FaBuilding}
          label={`Bairro: ${property.address.neighborhood}`}/>
      </li>}
  </List>
)