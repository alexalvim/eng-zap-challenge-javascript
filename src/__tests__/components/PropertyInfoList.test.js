import React from 'react';
import { shallow } from 'enzyme';

import { PORTALS } from '../../variables';
import PropertyInfoList from '../../components/PropertyInfoList';
import IconItem from '../../components/IconItem';

const mockedProperty = {
  bedrooms: 1,
  bathrooms: 1,
  parkingSpaces: 2,
  usableAreas: 10,
  address: {
    neighborhood: 'Moema'
  }
};

describe('Testing PropertyInfoList', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <PropertyInfoList
        property={mockedProperty}
        activePortal={PORTALS.ZAP}/>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should render with all properties', () => {
    const wrapper = shallow(
      <PropertyInfoList
        property={mockedProperty}
        activePortal={PORTALS.ZAP}/>
    );

    expect(wrapper.find(IconItem)).toHaveLength(5);
  });

  it('should render without neighborhood', () => {
    const wrapper = shallow(
      <PropertyInfoList
        property={{...mockedProperty, address: {}}}
        activePortal={PORTALS.ZAP}/>
    );

    expect(wrapper.find(IconItem)).toHaveLength(4);
  });

  it('should pass correct activePortal', () => {
    const wrapper = shallow(
      <PropertyInfoList
        property={{...mockedProperty, address: {}}}
        activePortal={PORTALS.ZAP}/>
    );

    expect(wrapper.find(IconItem).first().prop('activePortal')).toEqual(PORTALS.ZAP);
  });
});