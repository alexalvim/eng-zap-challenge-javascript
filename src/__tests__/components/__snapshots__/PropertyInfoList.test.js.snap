// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PropertyInfoList should render correctly 1`] = `ShallowWrapper {}`;
